### PR TITLE
Add Gitea Webhook

### PR DESCRIPTION
**Pull request by DblK**
While checking the webhook code, I saw that there were some gogs information.
So I fix the typo and the webhook test.

### CODE REVIEW

File: models/webhook.go
Comment (go2sh) [inline models/webhook.go]: Hm, you can add a line. But the copyright to the old stuff cannot be removed, FMPOV.

Comment (DblK) [reply models/webhook.go]: For other files, the copyright has been changed, so I did the same way but I do not care
I will wait for the review to be sure I should keep it or not.

Comment (lafriks) [reply models/webhook.go]: Only Gitea copyright is in files that are new and was not in Gogs. In places where new things are added new copyright line is added, see:
https://github.com/go-gitea/gitea/blob/a2019775903902622dc80c2231f105e3213f8a60/main.go#L2
Also it should be `The Gitea Authors`

**Comment by lafriks** [comment]
These changes will break backwards compatibility with existing integrations and API

File: models/webhook.go
Comment (bkcsoft) [reply models/webhook.go]: You should _add_ the copyright above the Gogs-line when changing something :)

**Comment by bkcsoft** [comment]
My biggest issue is that the Drone-integration will break, so I'd wait for https://github.com/drone/drone/issues/1978 first :)

**Comment by bkcsoft** [comment]
And we need a migration to fix all the current webhooks 🙄

**Comment by DblK** [comment]
There is a PR waiting for integration: https://github.com/drone/drone/pull/2017.
I agree that we should do it after the drone integration.

Good point for the migration. I won't do it right now because webhook might change. But once the drone PR is merged, I will work on it (if I have free time :p)

File: models/webhook.go
Comment (DblK) [reply models/webhook.go]: Okay, I took the example from this file https://github.com/go-gitea/gitea/blob/master/models/status.go#L1

**Comment by DblK** [comment]
Since https://github.com/drone/drone/pull/2017 has been merged, how we should finish this PR?

**Comment by lunny** [comment]
@Dblk, like @lafriks said, add a new webhook type `Gitea` is a good chosen but not change `Gogs` to `Gitea` so that we also have the ability to migrate Gogs' webhooks to Gitea when user migrate their's Gogs to Gitea.

**Comment by lunny** [comment]
Looks good.

**Comment by strk** [comment]
LGTM

**Comment by DblK** [comment]
Make L-G-T-M work!

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by appleboy** [comment]
make L-G-T-M work

**Comment by lunny** [comment]
@appleboy Please check the milestone before merge the PR.

**Comment by appleboy** [comment]
@lunny ok. my fault. I forgot to check the milestone.

**Pull request merged**
